### PR TITLE
IGNITE-21801 Add missing Schema Sync integration tests

### PR DESCRIPTION
**Pull request by rpuch**
https://issues.apache.org/jira/browse/IGNITE-21801

### CODE REVIEW

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/schemasync/ItSchemaSyncConfigTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+import org.apache.ignite.configuration.ConfigurationChangeException;
+import org.apache.ignite.configuration.validation.ConfigurationValidationException;
+import org.apache.ignite.internal.ClusterPerClassIntegrationTest;
+import org.apache.ignite.internal.app.IgniteImpl;
+import org.apache.ignite.internal.catalog.configuration.SchemaSynchronizationConfiguration;
+import org.junit.jupiter.api.Test;
+
+@SuppressWarnings("resource")
+class ItSchemaSyncConfigTest extends ClusterPerClassIntegrationTest {
```
Comment (tkalkirill) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/schemasync/ItSchemaSyncConfigTest.java:34]: I think the test class is redundant, the configuration framework should test this behavior and there is no need to double-check it in other modules.

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java [hunk 2]
```diff
@@ -1013,22 +1013,27 @@ private CompletableFuture<List<BinaryRow>> retrieveExactEntriesUntilCursorEmpty(
...
 
             return allOf(futs).thenApply((unused) -> rows);
         });
     }
 
+    private CompletableFuture<Void> validateBackwardCompatibility(BinaryRow row, UUID txId) {
+        return schemaCompatValidator.validateBackwards(row.schemaVersion(), tableId(), txId)
+                .thenAccept(validationResult -> {
+                    if (!validationResult.isSuccessful()) {
+                        throw new IncompatibleSchemaException(String.format(
```
Comment (tkalkirill) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java:1028]: As a suggestion, we could add a constructor that would accept an error message pattern and varags, inside use `org.apache.ignite.internal.lang.IgniteStringFormatter#format`.

Comment (rpuch) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java:1028]: I'm not sure such a change makes the situation so much better. The code at the exception creation site becomes just a bit shorter, but the constructor becomes more complex and now has an additional responsability. I would prefer to leave it in its current state.

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/schemasync/ItSchemaSyncConfigTest.java (hunk 1 above)
Comment (rpuch) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/schemasync/ItSchemaSyncConfigTest.java:34]: This is an integration test, and it seems usual to test something at a higher level, even if you already tested it using unit tests. Also, this property ('that delayDuration can never be changed') is very important, it's currently a cornerstone for the Schema Sync to work, so a little bit of redundancy here seems to be worth having it.

**Review by tkalkirill — APPROVED** [review]

**Pull request merged**